default/python/AI/AIstate.py

```diff
@@ -95,6 +147,14 @@ def __init__(self, aggression):
         self.empire_standard_enemy_rating = 0  # TODO: track on a per-empire basis
         self.character = create_character(aggression, self.empireID)
 
+    def __setstate__(self, state):
+        try:
+            for v in range(state.get("version", -1), AIstate.version):
+                convert_to_version(state, v+1)
+            self.__dict__ = state
+        except ConversionError:
+            self.__init__(aggression=fo.aggression.typical)
```

Comment:
An idea that just struck me here is that we could still double check if the provided state had an aggression attribute and then use that rather than necessarily defaulting to ```aggression.typical```.  That's probably actually the single most important piece of info from the savestate, and one of the least likely to be problematic.